### ISSUE TITLE
[3.5.1] Iterable is missing in lib.es2015.promise.d.ts

### ISSUE THREAD

**Issue by Rouden** [microsoft/TypeScript#31722]
**TypeScript Version:**  
* 3.5.1
* 3.6.0-dev.20190601

**Search Terms:** async, await, polyfill, csproj, TypeScriptLib 

**Code**

tsconfig.json
```json
{
  "compilerOptions": {
    "target": "es5",
    "module": "none",
    "lib": ["dom", "es5", "es2015.promise"],
    "strict": true,
  }
}
```

a.ts
```
// empty
```

**Expected behavior:**
```
> npx typescript@3.4.5
// ok
```

**Actual behavior:**
```
> npx typescript@3.5.1
../path/typescript/lib/lib.es2015.promise.d.ts:129:21 - error TS2304: Cannot find name 'Iterable'.

129     race<T>(values: Iterable<T>): Promise<T extends PromiseLike<infer U> ? U : T>;
                        ~~~~~~~~


Found 1 error.
```

**Playground Link:** None

**Related Issues:** Not found.

**Workaround:**
Add "es2015.iterable" to lib

**Comment by SpurYang** [comment]
Me too

**Comment by simonua** [comment]
Does this relate to #31676?

**Comment by SergioMorchon** [comment]
Any news on this?

**Comment by pushkin-** [comment]
@SergioMorchon I believe the linked issue is the same problem. Try using Typescript 3.5.3

**Comment by simonua** [comment]
3.5.3 exhibited the same problem for me. I have our TypeScript dependencies locked at 3.4.5 and cannot go higher at this time.

**Comment by SergioMorchon** [comment]
@pushkin- nope, with `3.5.3` shows the same error:
> node_modules/typescript/lib/lib.es2015.promise.d.ts:129:21 - error TS2304: Cannot find name 'Iterable'

**Comment by pushkin-** [comment]
@simonua , @SergioMorchon  Ok, I've deleted my comment.

**Comment by SergioMorchon** [comment]
@rbuckton any news on this? What do you recommend to bypass this issue to start using newer TypeScript versions?

**Comment by CervEdin** [comment]
This workaround appears to work

> Workaround is to add es2015.iterable to lib

https://github.com/headintheclouddev/typings-suitescript-2.0/issues/121#issuecomment-542797871

Unfortunately this ads typings for additional ES2015 iterable features

**Comment by rbuckton** [comment]
Fixed by #35998.

**Comment by rbuckton** [comment]
We are pushing `awaited` until after TS3.9 while we continue to assess the impact of the change. While this means we are essentially reverting #35998 with #37610, #37610 leaves in place the change that addressed this issue.
